projects/js-packages/connection/components/disconnect-survey/index.jsx

```diff
@@ -0,0 +1,187 @@
+	};
+
+	/**
+	 * Handle Submission of the survey.
+	 * Will send the survey response to the collection endpoint.
+	 */
+	const handleSurveySubmit = useCallback( () => {
+		const answerText = selectedAnswer === customOption.id ? customResponse : '';
+		onSubmit( selectedAnswer, answerText );
+	}, [ onSubmit, customOption.id, customResponse, selectedAnswer ] );
+
+	/**
+	 * Handle input into the custom response field.
+	 *
+	 * @param {object} e - onChange event for the custom input
+	 */
+	const handleCustomResponse = useCallback(
+		e => {
+			const value = e.target.value;
+			e.stopPropagation();
+			setCustomResponse( value );
+		},
+		[ setCustomResponse ]
+	);
+
+	/**
+	 * Checks to see if an option is the currently selected option, returns a css class name if it matches.
+	 *
+	 * @param {string} optionId   - ID of the option to check for.
+	 * @returns {string} - The "selected" class if this option is currently selected.
+	 */
+	const selectedClass = optionId => {
+		if ( optionId === selectedAnswer ) {
+			return 'jp-connect__disconnect-survey-card--selected';
+		}
+
+		return '';
+	};
+
+	/**
+	 * Event handler for keyboard events on the answer blocks.
+	 *
+	 * @param {string} answerId - The slug of the answer that has been selected.
+	 * @param {object} e - Keydown event.
+	 */
+	const handleAnswerKeyDown = useCallback(
+		( answerId, e ) => {
+			switch ( e.key ) {
+				case 'Enter':
+				case 'Space':
+				case 'Spacebar':
+				case ' ':
+					setSelectedAnswer( answerId );
+					break;
+			}
+		},
+		[ setSelectedAnswer ]
+	);
+
+	/**
+	 * Show all the survey options from the options array.
+	 *
+	 * @returns {React.ElementType []} - Mapped array of rendered survey options.
+	 */
+	const renderOptions = () => {
+		return options.map( option => {
+			return (
+				<SurveyChoice
+					id={ option.id }
+					onClick={ setSelectedAnswer }
+					onKeyDown={ handleAnswerKeyDown }
+					className={ 'card jp-connect__disconnect-survey-card ' + selectedClass( option.id ) }
+				>
+					<p className="jp-connect__disconnect-survey-card__answer">{ option.answerText }</p>
+				</SurveyChoice>
+			);
+		} );
+	};
+
+	/**
+	 * Show the custom input survey option.
+	 * Contains an input field for a custom response.
+	 *
+	 * @returns {React.ElementType} - The custom survey option with an input field.
+	 */
+	const renderCustomOption = () => {
+		return (
+			<SurveyChoice
+				id={ customOption.id }
+				onClick={ setSelectedAnswer }
+				onKeyDown={ handleAnswerKeyDown }
+				className={ 'card jp-connect__disconnect-survey-card ' + selectedClass( customOption.id ) }
+			>
+				<p className="jp-connect__disconnect-survey-card__answer">
+					{ __( 'Other:', 'jetpack' ) }{ ' ' }
+					<input
+						placeholder={ __( 'share your experience', 'jetpack' ) }
+						className="jp-connect__disconnect-survey-card__input"
+						type="text"
+						value={ customResponse }
+						onChange={ handleCustomResponse }
+						max={ 1000 } // Limit response length.
```

Comment:
While testing [20914](https://github.com/Automattic/jetpack/pull/20914), I noticed that this isn't limiting the characters to 1000 (is that the intention)? Might need `maxlength` instead?

Reply:
That is the intention here, yep! I used the wrong attribute by mistake 🤦  Will get this updated shortly.